core/bentley/src/FunctionChain.ts

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Utils
+ */
+
+function prepend(first: () => void, second?: () => void): () => void {
+  if (!second)
+    return first;
+
+  return () => {
+    first();
+    second();
+  };
+}
+
+/** Accumulates a series of functions to be executed in sequence at a later time.
+ * As an example, an object may want to register listeners for various [[BeEvent]]s, and unregister all of those listeners at some point in the future.
+ * The following code illustrates this usage:
+ * ```ts
+ * class Source {
+ *   public readonly onStringEvent = new BeEvent<string>();
+ *   public readonly onNumberEvent = new BeEvent<number>();
+ * }
+ *
+ * class Listener {
+ *   private readonly _unregisterListeners = new FunctionChain();
+ *
+ *   public constructor(source: Source) {
+ *     this._unregisterListeners.append(source.onStringEvent.addListener((str) => this.handleStringEvent(str)));
```

Comment:
For this example to work, you'd have to add `,this` to the addListener call.

Reply:
How so? Arrow functions capture `this`.